lnspec_py/basic_type/bitmask.py

```diff
@@ -56,3 +72,25 @@ def decode(hex_str: str) -> List[int]:
             if (bitfield[max_len // 8 - 1 - idx // 8] & (1 << (idx % 8))) != 0:
                 feature.append(idx)
         return feature
+
+    @staticmethod
+    def decode_with_len(hex_str: str) -> Tuple[BitfieldData, str]:
+        """
+        Deconding the bitfiled included the size of it by consuming the
+        input buffer.
+
+        params: hex_str the input hex string of the message
+        return: A touble compose by (BitfiledData, remains buffer)
+        """
+        # Take the first 4 hex digit (are 2 bytes) to decode the size f the array
+        size_buf = hex_str[:4]
+        hex_str = hex_str[4:]
+        lenght = u16Int(size_buf).decode()
```

Comment:
This type of operation can be done for all the type, so if we make this also of the integer type, we should have somethings like this 
```suggestion
        length, hex_str = u16Int(hex_str).decode()
```

where inside the u16Int decode method we do something like

```
def decode():
  size_buf = hex_str[:4]
   hex_str = hex_str[4:]
  # other decode op
  return self, hex_str
```

we should also refactoring the u16Int into U16Int and with the decode method that take the raw string in input so we avoid to store the input inside the class